File: src/main/java/nowcoder/Array2List.java
package nowcoder;

/**
 * @author LiuZhiguo
 * @date 2020/2/13 23:02
 */
public class Array2List {
    public ListNode array2List(int[] array){
        if (array.length == 0)
            return null;
        ListNode head = new ListNode(array[0]);
        ListNode first = head;
        for (int i=1;i<array.length;i++){
            ListNode node = new ListNode(array[i]);
            head.next = node;
            head = head.next;
        }
        return first;
    }
    public static void main(String[] args){
        int[] array = {1, 2, 3, 3, 4, 4, 5};
        Array2List array2List = new Array2List();
        ListNode head = array2List.array2List(array);
        while (head != null){
            System.out.println(head.val + " ");
            head = head.next;
        }
    }
}

File: src/main/Python/nowcoder/Array2List.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None
class Array2List:
    def array2List(self, array):
        if not array:
            return None
        head = ListNode(array[0])
        p = head
        for i in range(1, len(array)):
            node = ListNode(array[i])
            p.next = node
            p = p.next
        return head
if __name__ == '__main__':
    test = Array2List()
    array = [1, 2, 3, 4, 5]
    head = test.array2List(array)
    while head:
        print(head.val)
        head = head.next